### DISCUSSION TITLE
WiFi calling while abroad

### DISCUSSION THREAD

**Discussion by bige93** [kyujin-cho/pixel-volte-patch#63] [General]
Hi there! Just wanted to check what's your experience with using WiFi calling while outside of your home country. 
I use T-Mobile PL and VoWiFi in not registered while in Germany. It works in Poland. 
I tried with iPhone - WiFi calling connected right away.
Any similar experience?
![Screenshot_20230225-111950~2](https://user-images.githubusercontent.com/66921212/221362462-d30f6f2b-fa6f-430a-81ba-dd2083be0bfa.png)

Thanks!

**Comment by peixotorms** [comment]
did you try a polish vpn?

**Reply by bige93** [reply]
I didn't try that

**Reply by xantrk** [reply]
Same here. I am using dual SIM from different countries and neither of them connect to VoWIFI while roaming. I am sure at least one of them allows this. VPN doesn't do anything either.

**Comment by xantrk** [comment]
Same here. I am using dual SIM from different countries and neither of them connect to VoWIFI while roaming. I am sure at least one of them allows this. VPN doesn't do anything either.

**Comment by kszaq** [comment]
I have the same issue with Orange PL: as soon as SIM card is in "roaming" state, Pixel does not register to VoWiFi. The only workaround for now is to use Flight Mode with WiFi enabled, then I can call without issues. More importantly, I have a local eSIM that provides data and "backup calling" refuses to work because of roaming. I'm still looking if there's a flag that prevents VoWiFi registration while roaming.

**Reply by kszaq** [reply]
I might have found an answer to this issue... There is a flag called `carrier_default_wfc_ims_roaming_enabled_bool` - additionally, when you look up the Android code, most carriers have `carrier_supports_ss_over_ut_bool` enabled (https://cs.android.com/search?q=carrier_default_wfc_ims_roaming_enabled_bool). 

With these 2 flags in config override, it is working for me, both as "WiFi Calling" and "Backup calling":
```
mPersistentOverrideConfigs : 
        carrier_default_wfc_ims_enabled_bool = true
        carrier_default_wfc_ims_roaming_enabled_bool = true
        carrier_supports_ss_over_ut_bool = true
        carrier_volte_available_bool = true
        carrier_wfc_ims_available_bool = true
```

You can try an APK with this modifcation, remember to uninstall original version first because the signature won't match:
[app-debug.zip](https://github.com/kyujin-cho/pixel-volte-patch/files/10915718/app-debug.zip)
Source code patch:
[diff.txt](https://github.com/kyujin-cho/pixel-volte-patch/files/10915723/diff.txt)

How to use: reset carrier configuration to default in the app, then enable VoLTE (optional), VoWiFi but *do not* enable "Show VoWiFi Roaming Preference in Settings".

**Reply by xantrk** [reply]
This fixed it for me as well! Thanks for sharing this! Needed a reboot on top of everything but it did the trick!

Edit:  It looks like without rebooting I'm not able to get wifi call connected again (with disabling and reenabling). But still very good!

**Reply by kszaq** [reply]
I don't have issues with WiFi call not working after a reboot, I made several calls before/after rebooting and it worked fine.

I've been checking a little bit more and it looks like `carrier_supports_ss_over_ut_bool ` is not required, after all. The flag that does the trick is `carrier_default_wfc_ims_roaming_enabled_bool`:
```
mPersistentOverrideConfigs : 
        carrier_default_wfc_ims_enabled_bool = true
        carrier_default_wfc_ims_roaming_enabled_bool = true
        carrier_volte_available_bool = true
        carrier_wfc_ims_available_bool = true
        carrier_wfc_supports_wifi_only_bool = true
        hide_lte_plus_data_icon_bool = false
```
Here is the APK:
[app-debug2.zip](https://github.com/kyujin-cho/pixel-volte-patch/files/10920915/app-debug2.zip)
And diff:
[diff2.txt](https://github.com/kyujin-cho/pixel-volte-patch/files/10920919/diff2.txt)

I will PR these next week.
